messagequeue/messagequeue.go

```diff
@@ -64,27 +71,24 @@ func (mq *MessageQueue) AddMessage(entries []bsmsg.Entry, ses uint64) {
 
 // AddWantlist adds a complete session tracked want list to a message queue
 func (mq *MessageQueue) AddWantlist(initialWants *wantlist.SessionTrackedWantlist) {
-	mq.nextMessageLk.Lock()
-	defer mq.nextMessageLk.Unlock()
-
 	initialWants.CopyWants(mq.wl)
-	if initialWants.Len() > 0 {
-		if mq.nextMessage == nil {
-			mq.nextMessage = bsmsg.New(false)
-		}
-		for _, e := range initialWants.Entries() {
-			mq.nextMessage.AddEntry(e.Cid, e.Priority)
-		}
-		select {
-		case mq.outgoingWork <- struct{}{}:
-		default:
-		}
-	}
+	mq.addWantlist()
+}
+
+// SetRebroadcastInterval sets a new interval on which to rebroadcast the full wantlist
+func (mq *MessageQueue) SetRebroadcastInterval(delay time.Duration) {
+	mq.rebroadcastIntervalLk.Lock()
+	mq.rebroadcastInterval = delay
+	mq.rebroadcastTimer.Reset(delay)
```

Comment:
We should probably check if `rebroadcastTimer` is nil here.

Reply:
good point.